torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""EuroCrops dataset."""
+
+import csv
+import os
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import fiona
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import VectorDataset
+from .utils import (
+    DatasetNotFoundError,
+    check_integrity,
+    download_and_extract_archive,
+    download_url,
+)
+
+
+class EuroCrops(VectorDataset):
+    """EuroCrops Dataset (Version 9).
+
+    The `EuroCrops <https://www.eurocrops.tum.de/index.html>`__ dataset combines "all
+    publicly available self-declared crop reporting datasets from countries of the
+    European Union" into a unified format. The dataset is released under CC BY 4.0 Deed.
```

Comment:
Here we say CC-BY-4.0 (because that's what the website says), but our docs say CC-BY-SA-4.0 (because that's what GitHub/Zenodo use). Not sure which to trust, but maybe we should be consistent.